rankers/LightGBMRanker/__init__.py

```diff
@@ -0,0 +1,111 @@
+import os
+from typing import Dict, Optional, Tuple, List, Union
+
+import numpy as np
+
+from jina.executors.decorators import batching_multi_input
+from jina.executors.rankers import Match2DocRanker
+from jina.excepts import PretrainedModelFileDoesNotExist
+
+if False:
+    import lightgbm
+
+
+class LightGBMRanker(Match2DocRanker):
+    """
+    Computes a relevance score for each match using a pretrained Ltr model trained with LightGBM (https://lightgbm.readthedocs.io/en/latest/index.html)
+
+    :param model_path: path to the pretrained model previously trained using LightGBM
+    :param query_feature_names: name of the features to extract from query Documents and used to compute relevance scores by the model loaded
+    from model_path
+    :param match_feature_names: name of the features to extract from match Documents and used to compute relevance scores by the model loaded
+    from model_path
+    :param query_categorical_features: name of features contained in `query_feature_names` corresponding to categorical features.
+    :param match_categorical_features: name of features contained in `match_feature_names` corresponding to categorical features.
+    :param query_features_before: True if `query_feature_names` must be placed before the `match` ones in the `dataset` used for prediction.
+    :param args: Additional positional arguments
+    :param kwargs: Additional keyword arguments
+
+    .. note::
+        The name of the features are used to extract the features from incoming `documents`. Check how these features are accessed in
+        :class:`Document`
+
+    """
+
+    def __init__(self,
+                 model_path: Optional[str] = 'tmp/model.txt',
+                 query_feature_names: Tuple[str] = ['tags__query_length', 'tags__query_language'],
+                 match_feature_names: Tuple[str] = ['tags__document_length', 'tags__document_language',
+                                                    'tags__document_pagerank'],
```

Comment:
I'm not sure whether the default values are proper. At least, we need to mention the default values in the docstring.
```suggestion
                 query_feature_names: Tuple[str] = ['tags__query_length', 'tags__query_language'],
                 match_feature_names: Tuple[str] = ['tags__document_length', 'tags__document_language',
                                                    'tags__document_pagerank'],
```

Reply:
there are no really good default values